# HG changeset patch
# User Kirill Belokurov <kirill.belokurov@gmail.com>
# Date 1206712354 -7200
# Node ID 70e9598c5dd820ff98556352e2c1ed038ade2224
# Parent  39383740e5296a82181f4857400d3f22f49f94fd
AIFF comment chunks are word-aligned, so we should seek extra byte for them

diff -r 39383740e529 -r 70e9598c5dd8 src/demuxers/demux_aiff.c
--- a/src/demuxers/demux_aiff.c	Thu Mar 27 17:29:00 2008 +0200
+++ b/src/demuxers/demux_aiff.c	Fri Mar 28 15:52:34 2008 +0200
@@ -46,6 +46,10 @@
 #define COMM_TAG FOURCC_TAG('C', 'O', 'M', 'M')
 #define SSND_TAG FOURCC_TAG('S', 'S', 'N', 'D')
 #define APCM_TAG FOURCC_TAG('A', 'P', 'C', 'M')
+#define NAME_TAG FOURCC_TAG('N', 'A', 'M', 'E')
+#define AUTH_TAG FOURCC_TAG('A', 'U', 'T', 'H')
+#define COPY_TAG FOURCC_TAG('(', 'c', ')', ' ')
+#define ANNO_TAG FOURCC_TAG('A', 'N', 'N', 'O')
 
 #define PREAMBLE_SIZE 8
 #define AIFF_SIGNATURE_SIZE 12
@@ -177,6 +181,11 @@ static int open_aiff_file(demux_aiff_t *
       break;
 
     } else {
+      /* if chunk contains metadata, it will be word-aligned (as seen at sox and ffmpeg) */
+      if ( ((chunk_type == NAME_TAG) || (chunk_type==AUTH_TAG) ||
+            (chunk_type == COPY_TAG) || (chunk_type==ANNO_TAG)) && (chunk_size&1))
+	chunk_size++;
+
       /* unrecognized; skip it */
       this->input->seek(this->input, chunk_size, SEEK_CUR);
     }
